src/services/download/mod.rs

```diff
@@ -204,6 +201,25 @@ fn download_future_stream(
     Box::new(ret)
 }
 
+async fn download_stream(
+    source: SourceFileId,
+    stream: impl futures::stream::Stream<Item = Result<bytes::Bytes, DownloadError>>,
+    destination: PathBuf,
+) -> Result<DownloadStatus, DownloadError> {
+    // All file I/O in this function is blocking!
+    log::trace!("Downloading from {}", source);
+    let mut file =
+        std::fs::File::create(&destination).context(DownloadErrorKind::BadDestination)?;
```

Comment:
Please import `Bytes`, `Stream` and `File.` For `Stream`, alias the old one to `Stream01`.